src/main/java/uk/gov/companieshouse/api/accounts/service/impl/FilingServiceImpl.java

```diff
@@ -0,0 +1,383 @@
+package uk.gov.companieshouse.api.accounts.service.impl;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.api.accounts.AccountsType;
+import uk.gov.companieshouse.api.accounts.LinkType;
+import uk.gov.companieshouse.api.accounts.model.entity.CompanyAccountEntity;
+import uk.gov.companieshouse.api.accounts.model.filing.Data;
+import uk.gov.companieshouse.api.accounts.model.filing.Filing;
+import uk.gov.companieshouse.api.accounts.model.filing.Link;
+import uk.gov.companieshouse.api.accounts.model.ixbrl.Account;
+import uk.gov.companieshouse.api.accounts.service.FilingService;
+import uk.gov.companieshouse.api.accounts.transaction.Transaction;
+import uk.gov.companieshouse.api.accounts.util.ixbrl.accountsbuilder.AccountsBuilder;
+import uk.gov.companieshouse.api.accounts.util.ixbrl.ixbrlgenerator.DocumentGeneratorConnection;
+import uk.gov.companieshouse.api.accounts.util.ixbrl.ixbrlgenerator.IxbrlGenerator;
+import uk.gov.companieshouse.document.data.DocumentDescriptionHelper;
+import uk.gov.companieshouse.environment.EnvironmentReader;
+import uk.gov.companieshouse.environment.impl.EnvironmentReaderImpl;
+
+@Service
+public class FilingServiceImpl implements FilingService {
+
+    private static final String API_KEY_ENV_VAR = "CHS_API_KEY";
```

Comment:
I would use the environment reader here like we have used in the Mongo properties, so the variable is loaded once when the class is initialised

Reply:
Additionally, I would do the same on all other env variables

Reply:
I wont add load the environment variables into member variables when the class is loaded as I have junit issues if I use this approach.  Therefore, I will get the environmnet variable value only when it is needed.